pkg/datapath/loader/cache.go

```diff
@@ -72,6 +109,111 @@ func (o *ObjectCache) Update(nodeCfg *datapath.LocalNodeConfiguration) {
 	o.baseHash = newHash
 }
 
+// serialize finds the channel that serializes builds against the same hash.
+// Returns the channel and whether or not the caller needs to compile the
+// datapath for this hash.
+func (o *ObjectCache) serialize(hash string) (fq *serializer.FunctionQueue, found bool) {
+	o.Lock()
+	defer o.Unlock()
+
+	fq, compiled := o.compileQueue[hash]
+	if !compiled {
+		fq = serializer.NewFunctionQueue(1)
+		o.compileQueue[hash] = fq
+	}
+	return fq, compiled
+}
+
+func (o *ObjectCache) lookup(hash string) (string, bool) {
+	o.Lock()
+	defer o.Unlock()
```

Comment:
remove the defer and add the unlock on line 130-131

Reply:
Can you explain why? I don't understand the difference.